great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -11,4 +46,459 @@ class AbstractDataContext(ABC):
+            logger.info(
+                "Profiler store is not configured; omitting it from active stores"
+            )
+
+        return [
+            store for store in self.list_stores() if store["name"] in active_store_names
+        ]
+
+    # private methods
+
+    def _build_store_from_config(
+        self,
+        store_name: str,
+        store_config: dict,
+        runtime_environment: Optional[dict] = None,
+    ) -> Optional[Store]:
+        module_name = "great_expectations.data_context.store"
+        # Set expectations_store.store_backend_id to the data_context_id from the project_config if
+        # the expectations_store does not yet exist by:
+        # adding the data_context_id from the project_config
+        # to the store_config under the key manually_initialize_store_backend_id
+        if (store_name == self.expectations_store_name) and store_config.get(
+            "store_backend"
+        ):
+            store_config["store_backend"].update(
+                {
+                    "manually_initialize_store_backend_id": self.project_config_with_variables_substituted.anonymous_usage_statistics.data_context_id
+                }
+            )
+
+        # Set suppress_store_backend_id = True if store is inactive and has a store_backend.
+        if (
+            store_name not in [store["name"] for store in self.list_active_stores()]
+            and store_config.get("store_backend") is not None
+        ):
+            store_config["store_backend"].update({"suppress_store_backend_id": True})
+
+        new_store = build_store_from_config(
+            store_name=store_name,
+            store_config=store_config,
+            module_name=module_name,
+            runtime_environment=runtime_environment,
+        )
+        self._stores[store_name] = new_store
+        return new_store
+
+    def _construct_data_context_id(self) -> str:
+        """
+        Choose the id of the currently-configured expectations store, if available and a persistent store.
+        If not, it should choose the id stored in DataContextConfig.
+        Returns:
+            UUID to use as the data_context_id
+        """
+        # Choose the id of the currently-configured expectations store, if it is a persistent store
+        expectations_store = self._stores[
+            self.project_config_with_variables_substituted.expectations_store_name
+        ]
+        if isinstance(expectations_store.store_backend, TupleStoreBackend):
+            # suppress_warnings since a warning will already have been issued during the store creation if there was an invalid store config
+            return expectations_store.store_backend_id_warnings_suppressed
+
+        # Otherwise choose the id stored in the project_config
+        else:
+            return (
+                self.project_config_with_variables_substituted.anonymous_usage_statistics.data_context_id
+            )
+
+    def _load_config_variables_file(self):
```

Comment:
I'd rather see if we can eliminate this method (ie have it exist only in FileDataContext)